corehq/apps/app_manager/fields.py

```diff
@@ -116,3 +137,387 @@ def get_app_sources(domain):
         }
         for app in apps
     }
+
+
+class ApplicationDataRMIHelper(object):
+    """ApplicationDataRMIHelper is meant to generate the response for the
+    djangoRMI methods required to initialize form controlled by
+    hq.app_data_drilldown.ng.js.
+
+    Note / todo: This Helper should be merged with ApplicationDataSourceUIHelper.
+    Holding off to a different PR, as I want to isolate QA to just the exports
+    (the first thing to use this) --Biyeun
+
+    """
+    UNKNOWN_SOURCE = '_unknown'
+
+    APP_TYPE_ALL = 'all'
+    APP_TYPE_DELETED = 'deleted'
+    APP_TYPE_REMOTE = 'remote'
+    APP_TYPE_NONE = 'no_app'
+
+    def __init__(self, domain, as_dict=True, form_placeholders=None, case_placeholders=None):
+        self.domain = domain
+        self.as_dict = as_dict
+        default_form_placeholder = AppFormRMIPlaceholder(
+            application=_("Select Application"),
+            module=_("Select Module"),
+            form=_("Select Form"),
+        )
+        self.form_placeholders = form_placeholders or default_form_placeholder
+        default_case_placeholder = AppCaseRMIPlaceholder(
+            application=_("Select Application"),
+            case_type=_("Select Case Type"),
+        )
+        self.case_placeholders = case_placeholders or default_case_placeholder
+        if self.as_dict:
+            self.form_placeholders = self.form_placeholders._asdict()
+            self.case_placeholders = self.case_placeholders._asdict()
+
+    def _get_form_results(self):
+        return XFormInstance.get_db().view(
```

Comment:
ok will make a few small follow-up PRs to address this before lifting the feature flag.
http://manage.dimagi.com/default.asp?187459